### PR TITLE
feat(openapi): entity endpoints & analytics raw

### PR DESCRIPTION
**Pull request by david-leifker**
* Expand OpenAPI with core entities and a few aspects
* Raw experimental OpenAPI endpoint to DataHub Usage Events

Screenshot 2023-08-08 at 7 04 54 AM

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Comment by david-leifker** [comment]
Ran locally needs tests, looking for early feedback.

File: metadata-service/openapi-entity-servlet/src/main/java/io/datahubproject/openapi/delegates/EntityApiDelegateImpl.java [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+
+    public ResponseEntity<GlossaryTermsAspectResponseV2> createGlossaryTerms(GlossaryTermsAspectRequestV2 body, String urn) {
+        String methodName = walker.walk(frames -> frames
+                .findFirst()
+                .map(StackWalker.StackFrame::getMethodName)).get();
+        return createAspect(urn, methodNameToAspectName(methodName), body, GlossaryTermsAspectRequestV2.class, GlossaryTermsAspectResponseV2.class);
+    }
+
+    public ResponseEntity<OwnershipAspectResponseV2> createOwnership(OwnershipAspectRequestV2 body, String urn) {
+        String methodName = walker.walk(frames -> frames
```
Comment (RyanHolstien) [inline metadata-service/openapi-entity-servlet/src/main/java/io/datahubproject/openapi/delegates/EntityApiDelegateImpl.java:144]: This is cool 😄 looks like it maintains thread context too so concurrency doesn't mess with it, nice!

**Review by RyanHolstien — COMMENTED** [review]
Probably worth having a Zoom review/demo over this one if you have bandwidth, there's a lot going on esp in the OpenApiEntities class. I think all the changes to the existing logic make sense to me though.

File: metadata-service/openapi-entity-servlet/src/main/java/io/datahubproject/openapi/delegates/EntityApiDelegateImpl.java [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+
+    public ResponseEntity<BrowsePathsV2AspectResponseV2> createBrowsePathsV2(@Valid BrowsePathsV2AspectRequestV2 body, String urn) {
+        String methodName = walker.walk(frames -> frames
+                .findFirst()
+                .map(StackWalker.StackFrame::getMethodName)).get();
+        return createAspect(urn, methodNameToAspectName(methodName), body, BrowsePathsV2AspectRequestV2.class,
+                BrowsePathsV2AspectResponseV2.class);
+    }
+
+    public ResponseEntity<S> scroll(@Valid Boolean systemMetadata, @Valid List<String> aspects, @Min(1) @Valid Integer count,
```
Comment (david-leifker) [inline metadata-service/openapi-entity-servlet/src/main/java/io/datahubproject/openapi/delegates/EntityApiDelegateImpl.java:347]: Check authorization

Comment (david-leifker) [reply metadata-service/openapi-entity-servlet/src/main/java/io/datahubproject/openapi/delegates/EntityApiDelegateImpl.java:347]: Check policy authorization

File: metadata-service/openapi-analytics-servlet/src/main/java/io/datahubproject/openapi/config/OpenapiAnalyticsConfig.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package io.datahubproject.openapi.config;
```
Comment (david-leifker) [inline metadata-service/openapi-analytics-servlet/src/main/java/io/datahubproject/openapi/config/OpenapiAnalyticsConfig.java:1]: Tie in authorization for analytics, use-analytics-api perm

**Comment by david-leifker** [comment]
Add test for custom entity.

**Review by shirshanka — APPROVED** [review]
LGTM

**Pull request merged**